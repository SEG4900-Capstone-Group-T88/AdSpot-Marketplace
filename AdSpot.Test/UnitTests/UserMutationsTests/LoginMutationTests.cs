namespace AdSpot.Test.UnitTests.UserMutationsTests;

public class LoginMutationTests
{
    private const string LoginMutation = """
        mutation LoginMutation($input: LoginInput!) {
            login(input: $input) {
                user {
                    userId
                    email
                    firstName
                    lastName
                }

                errors {
                    ... on Error {
                        __typename
                        message
                    }
                }
            }
        }
        """;

    [Fact]
    [Trait("Category", "Unit")]
    public async Task LoginSuccessful()
    {
        var result = await TestServices.ExecuteRequestAsync(b =>
            b.SetQuery(LoginMutation)
                .SetVariableValue(
                    "input",
                    new Dictionary<string, object?> { { "email", "user1" }, { "password", "user1" }, }.AsReadOnly()
                )
        );

        result.MatchSnapshot();
    }

    [Fact]
    [Trait("Category", "Unit")]
    public async Task LoginInvalidCredentials()
    {
        var result = await TestServices.ExecuteRequestAsync(b =>
            b.SetQuery(LoginMutation)
                .SetVariableValue(
                    "input",
                    new Dictionary<string, object?>
                    {
                        { "email", "user1" },
                        { "password", "invalidcredentials" },
                    }.AsReadOnly()
                )
        );

        result.MatchSnapshot();
    }

    [Fact]
    [Trait("Category", "Unit")]
    public async Task LoginUserDoesNotExist()
    {
        var result = await TestServices.ExecuteRequestAsync(b =>
            b.SetQuery(LoginMutation)
                .SetVariableValue(
                    "input",
                    new Dictionary<string, object?>
                    {
                        { "email", "userdoesnotexist" },
                        { "password", "userdoesnotexist" },
                    }.AsReadOnly()
                )
        );

        result.MatchSnapshot();
    }
}
