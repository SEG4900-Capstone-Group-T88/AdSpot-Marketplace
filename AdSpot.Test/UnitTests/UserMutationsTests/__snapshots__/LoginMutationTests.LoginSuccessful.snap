{
  "data": {
    "login": {
      "user": {
        "userId": 1,
        "email": "testuser@adspot.com",
        "firstName": "Test",
        "lastName": "User"
      },
      "errors": null
    }
  }
}